### PR TITLE
add multivariate test sample

### PR DESCRIPTION
**Pull request by luxiong-ms**
# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/master/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

File: sdk/anomalydetector/Azure.AI.AnomalyDetector/tests/samples/Sample4_MultivariateDetect.cs [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+using System.Text;
+using System.Threading.Tasks;
+using Azure.AI.AnomalyDetector.Models;
+using Azure.Core.TestFramework;
+using Microsoft.Identity.Client;
+using NUnit.Framework;
+
+namespace Azure.AI.AnomalyDetector.Tests.Samples
+{
+    public partial class AnomalyDetectorSamples : SamplesBase<AnomalyDetectorTestEnvironment>
```
Comment (ShivangiReja) [inline sdk/anomalydetector/Azure.AI.AnomalyDetector/tests/samples/Sample4_MultivariateDetect.cs:21]: Can we add an md file for this sample with the description like other three samples [here](https://github.com/Azure/azure-sdk-for-net/tree/7c71881e7406c63a17de67c7b52ee1be67648d9d/sdk/anomalydetector/Azure.AI.AnomalyDetector/samples)?

Comment (ShivangiReja) [reply sdk/anomalydetector/Azure.AI.AnomalyDetector/tests/samples/Sample4_MultivariateDetect.cs:21]: I still don't see an md file [here](https://github.com/Azure/azure-sdk-for-net/tree/7c71881e7406c63a17de67c7b52ee1be67648d9d/sdk/anomalydetector/Azure.AI.AnomalyDetector/samples) for `Sample4_MultivariateDetect.cs` sample. We need to create a `Sample4_MultivariateDetect.md` file like [this](https://github.com/Azure/azure-sdk-for-net/blob/7c71881e7406c63a17de67c7b52ee1be67648d9d/sdk/anomalydetector/Azure.AI.AnomalyDetector/samples/Sample1_DetectEntireSeriesAnomaly.md) with the description explaining the sample.

**Comment by ShivangiReja** [comment]
@scbedd is looking into the pipeline failure in `Link verification check` step because [these](https://dev.azure.com/azure-sdk/public/_build/results?buildId=855903&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=bdeefc16-b669-5ebd-ad94-a2c19ade53b0&l=20) links work fine.

File: sdk/anomalydetector/Azure.AI.AnomalyDetector/tests/samples/Sample4_MultivariateDetect.cs [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+                }
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(String.Format("Export error. {0}", e.Message));
+                throw new Exception(e.Message);
+            }
+        }
+
+        private async Task deleteAsync(AnomalyDetectorClient client, Guid model_id)
```
Comment (ShivangiReja) [inline sdk/anomalydetector/Azure.AI.AnomalyDetector/tests/samples/Sample4_MultivariateDetect.cs:182]: Add snippet region here. Check [this](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/anomalydetector/Azure.AI.AnomalyDetector/tests/samples/Sample1_DetectEntireSeriesAnomaly.cs#L22) for an example. You can also refer the docs [here](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#updating-sample-snippets).

File: sdk/anomalydetector/Azure.AI.AnomalyDetector/tests/samples/Sample4_MultivariateDetect.cs [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+                return result.Value;
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(String.Format("Detection error. {0}", e.Message));
+                throw new Exception(e.Message);
+            }
+        }
+
+        private async Task exportAsync(AnomalyDetectorClient client, Guid model_id, string model_path = "model.zip")
```
Comment (ShivangiReja) [inline sdk/anomalydetector/Azure.AI.AnomalyDetector/tests/samples/Sample4_MultivariateDetect.cs:162]: Add snippet region here. Check [this](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/anomalydetector/Azure.AI.AnomalyDetector/tests/samples/Sample1_DetectEntireSeriesAnomaly.cs#L22) for an example. You can also refer the docs [here](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#updating-sample-snippets).

File: sdk/anomalydetector/Azure.AI.AnomalyDetector/tests/samples/Sample4_MultivariateDetect.cs [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+                Console.WriteLine(String.Format("{0} available models after training.", model_number));
+                return trained_model_id;
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(String.Format("Train error. {0}", e.Message));
+                throw new Exception(e.Message);
+            }
+        }
+        private async Task<DetectionResult> detectAsync(AnomalyDetectorClient client, string datasource, Guid model_id,DateTimeOffset start_time, DateTimeOffset end_time, int max_tryout = 500)
```
Comment (ShivangiReja) [inline sdk/anomalydetector/Azure.AI.AnomalyDetector/tests/samples/Sample4_MultivariateDetect.cs:126]: Add snippet region.

File: sdk/anomalydetector/Azure.AI.AnomalyDetector/tests/samples/Sample4_MultivariateDetect.cs [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+                String msg = String.Format("Multivariate error. {0}", e.Message);
+                if (model_id_raw != null)
+                {
+                    await deleteAsync(client, model_id_raw.GetValueOrDefault()).ConfigureAwait(false);
+                }
+                Console.WriteLine(msg);
+                throw new Exception(msg);
+            }
+        }
+        private async Task<Guid?> trainAsync(AnomalyDetectorClient client, string datasource, DateTimeOffset start_time, DateTimeOffset end_time, int max_tryout = 500)
```
Comment (ShivangiReja) [inline sdk/anomalydetector/Azure.AI.AnomalyDetector/tests/samples/Sample4_MultivariateDetect.cs:76]: Add snippet region.

**Review by ShivangiReja — CHANGES_REQUESTED** [review]

**Comment by ShivangiReja** [comment]
@luxiong-ms - While @scbedd is looking into the broken links issue. I added the links in `eng/ignore-links.txt` file to unblock this PR.

Added few comments. Please take a look.

File: sdk/anomalydetector/Azure.AI.AnomalyDetector/samples/Sample4_MultivariateDetect.md [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+
+## Train the model
+
+Create a new private async task as below to handle training your model. You will use `TrainMultivariateModel` to train the model and `GetMultivariateModelAysnc` to check when training is complete.
+
+You could add the data source, along with start time and end time to a `ModelInfo` object to create a data feed. The data source is a shared access signature(SAS) link in the format https://\[placeholder\]/. To generate the datasource link, you could first download our [sample data][datasource], then upload it to a azure container according to the [Upload a block blob][upload_blob] documentation and get the SAS link of the data according to the [Create SAS tokens for blobs in the Azure portal][generate_sas] documentation.
+
+Call `TrainMultivariateModel` with the created data feed and extract the model ID from the response header. Afterwards, you can get the model info, including the model status, by calling `GetMultivariateModelAsync` with the model ID . Wait until the model status is ready. 
+
+```C# Snippet:TrainModel
```
Comment (ShivangiReja) [inline sdk/anomalydetector/Azure.AI.AnomalyDetector/samples/Sample4_MultivariateDetect.md:32]: Snippet region name is different. It should be the same as `TrainMultivariateModel`

Comment (ShivangiReja) [reply sdk/anomalydetector/Azure.AI.AnomalyDetector/samples/Sample4_MultivariateDetect.md:32]: Please fix this for other snippets as well.

File: sdk/anomalydetector/Azure.AI.AnomalyDetector/tests/samples/Sample4_MultivariateDetect.cs [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+            {
+                String msg = String.Format("Multivariate error. {0}", e.Message);
+                if (model_id_raw != null)
+                {
+                    await deleteAsync(client, model_id_raw.GetValueOrDefault()).ConfigureAwait(false);
+                }
+                Console.WriteLine(msg);
+                throw new Exception(msg);
+            }
+        }
```
Comment (ShivangiReja) [inline sdk/anomalydetector/Azure.AI.AnomalyDetector/tests/samples/Sample4_MultivariateDetect.cs:75]: nit: skip line after the method

**Review by ShivangiReja — CHANGES_REQUESTED** [review]

**Comment by ShivangiReja** [comment]
@luxiong-ms

**Comment by ShivangiReja** [comment]
@luxiong-ms 

**[error]Generated code is not up to date.**

 You may need to rebase on the latest master, 
 * run `eng\scripts\Update-Snippets.ps1` if you modified sample snippets or other *.md files (https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#updating-sample-snippets), 
 * run `eng\scripts\Export-API.ps1` if you changed public APIs (https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#public-api-additions). 
 * run `dotnet build /t:GenerateCode` to update the generated code.

File: eng/ignore-links.txt [hunk 8]
```diff
@@ -5,3 +5,5 @@ https://github.com/Azure/azure-sdk-for-net-pr/tree/feature/IoT-ADT/sdk/digitaltw
 https://github.com/Azure/azure-sdk-for-net-pr/blob/feature/IoT-ADT/sdk/iot/Azure.Iot.DigitalTwins/samples/DigitalTwinServiceClientSample/DTDL/Relationships/HospitalRelationships.json
 https://www.nuget.org/packages/Azure.ResourceManager.DigitalTwins
 https://contoso.azureedge.net/urlsigning/test?expires=2145916800&amp;keyid:key1&amp;signature=iTsrLX9rVAIJkSahBA_j5o9Azf5-j331ohxDR1Gx2js=
+https://docs.microsoft.com/en-us/azure/storage/blobs/storage-quickstart-blobs-portal
```
Comment (weshaggard) [inline eng/ignore-links.txt:8]: Why are we adding these links to the ignore list? We should not have broken links in our readmes. See http://aka.ms/azsdk/guideline/links.

Comment (ShivangiReja) [reply eng/ignore-links.txt:8]: `https://docs.microsoft.com/en-us/azure/storage/blobs/storage-quickstart-blobs-portal` is not a broken link but the pipeline was failing with broken links error(check [here](https://dev.azure.com/azure-sdk/public/_build/results?buildId=855903&view=logs&j=36df4387-3551-57fa-098c-b43a23930261&t=3851d55a-4253-57f5-016e-927e8de9d21b&l=19)). I contacted @scbedd, he said 
>I'm going to need to follow up with the script writers. I've been looking at this backwards and forwards and I still don't know why it's failing the check.

To unblock this PR he suggested to add these links in `eng/ignore-links.txt` file while he is looking into the issue.

Comment (weshaggard) [reply eng/ignore-links.txt:8]:
```
##[warning]DO NOT include locale /en-us/ information in links: https://docs.microsoft.com/en-us/azure/storage/blobs/storage-quickstart-blobs-portal. Check here for more information: https://aka.ms/azsdk/guideline/links
##[warning]DO NOT include locale /en-us/ information in links: https://docs.microsoft.com/en-us/azure/cognitive-services/translator/document-translation/create-sas-tokens?tabs=Containers. Check here for more information: https://aka.ms/azsdk/guideline/links
```

The warning tells you what to do and provide a link for more information.

Comment (luxiong-ms) [reply eng/ignore-links.txt]: Thanks for the advice. The /en-us/ is removed from the links.

**Review by ShivangiReja — APPROVED** [review]

**Pull request merged**